sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FilePartitionReader.scala

```diff
@@ -72,7 +73,9 @@ class FilePartitionReader[T](readers: Iterator[PartitionedFileReader[T]])
           s"Expected: ${e.getLogicalType}, Found: ${e.getPhysicalType}"
         throw new QueryExecutionException(message, e)
       case e: ParquetDecodingException =>
-        if (e.getMessage.contains("Can not read value at")) {
+        if (e.getCause.isInstanceOf[SparkUpgradeException]) {
+          throw e.getCause
```

Comment:
If this is user-facing is it possible that it returns `null`? If it returns `null` message won't be helpful.

Reply:
In practice, this never happens since `ParquetDecodingException` always wraps an exception there https://github.com/apache/parquet-mr/blob/ee30b13bb5c3f6848c76641d3b93c9858e6746cb/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordReader.java#L253-L255